blivet/tasks/availability.py

```diff
@@ -246,15 +285,108 @@ def available_resource(name):
+                           blockdev.LoopTechMode.DESTROY |
+                           blockdev.LoopTechMode.MODIFY |
+                           blockdev.LoopTechMode.QUERY)
+BLOCKDEV_LOOP = BlockDevTechInfo(plugin_name="loop",
+                                 check_fn=blockdev.loop_is_tech_avail,
+                                 technologies={blockdev.LoopTech.LOOP_TECH_LOOP: BLOCKDEV_LOOP_ALL_MODES})  # FIXME
+BLOCKDEV_LOOP_TECH = BlockDevMethod(BLOCKDEV_LOOP)
+
+# libblockdev lvm plugin required technologies and modes
+BLOCKDEV_LVM_ALL_MODES = (blockdev.LVMTechMode.CREATE |
+                          blockdev.LVMTechMode.REMOVE |
+                          blockdev.LVMTechMode.MODIFY |
+                          blockdev.LVMTechMode.QUERY)
+BLOCKDEV_LVM = BlockDevTechInfo(plugin_name="lvm",
+                                check_fn=blockdev.lvm_is_tech_avail,
+                                technologies={blockdev.LVMTech.BASIC: BLOCKDEV_LVM_ALL_MODES,
+                                              blockdev.LVMTech.BASIC_SNAP: BLOCKDEV_LVM_ALL_MODES,
+                                              blockdev.LVMTech.THIN: BLOCKDEV_LVM_ALL_MODES,
+                                              blockdev.LVMTech.CACHE: BLOCKDEV_LVM_ALL_MODES,
+                                              blockdev.LVMTech.CALCS: blockdev.LVMTechMode.QUERY,
+                                              blockdev.LVMTech.THIN_CALCS: blockdev.LVMTechMode.QUERY,
+                                              blockdev.LVMTech.CACHE_CALCS: blockdev.LVMTechMode.QUERY,
+                                              blockdev.LVMTech.GLOB_CONF: blockdev.LVMTechMode.QUERY})
+BLOCKDEV_LVM_TECH = BlockDevMethod(BLOCKDEV_LVM)
+
+# libblockdev mdraid plugin required technologies and modes
+BLOCKDEV_MD_ALL_MODES = (blockdev.MDTechMode.CREATE |
+                         blockdev.MDTechMode.DELETE |
+                         blockdev.MDTechMode.MODIFY |
+                         blockdev.MDTechMode.QUERY)
+BLOCKDEV_MD = BlockDevTechInfo(plugin_name="mdraid",
+                               check_fn=blockdev.md_is_tech_avail,
+                               technologies={blockdev.MDTech.MD_TECH_MDRAID: BLOCKDEV_MD_ALL_MODES})
+BLOCKDEV_MD_TECH = BlockDevMethod(BLOCKDEV_MD)
+
+# libblockdev mpath plugin required technologies and modes
+BLOCKDEV_MPATH_ALL_MODES = (blockdev.MpathTechMode.MODIFY |
+                            blockdev.MpathTechMode.QUERY)
+BLOCKDEV_MPATH = BlockDevTechInfo(plugin_name="mpath",
+                                  check_fn=blockdev.mpath_is_tech_avail,
+                                  technologies={blockdev.MpathTech.BASE: BLOCKDEV_MPATH_ALL_MODES})
+BLOCKDEV_MPATH_TECH = BlockDevMethod(BLOCKDEV_MPATH)
+
+# libblockdev swap plugin required technologies and modes
+BLOCKDEV_SWAP_ALL_MODES = (blockdev.SwapTechMode.CREATE |
+                           blockdev.SwapTechMode.ACTIVATE_DEACTIVATE |
+                           blockdev.SwapTechMode.QUERY)
```

Comment:
These are not all modes of the swap tech.

Reply:
Yes, but these are all modes we need in blivet. (We don't yet support changing label on swap.)